src/main/java/org/ow2/proactive/connector/iaas/cloud/provider/azure/scaleset/AzureScaleSetProvider.java

```diff
@@ -0,0 +1,382 @@
+                                                 .map(publicIpAddresses -> azureProviderUtils.searchPublicIpAddressByIp(azureService,
+                                                                                                                        publicIpAddresses)
+                                                                                             .get())
+                                                 .orElse(azureService.publicIpAddresses()
+                                                                     .define(azureIPName)
+                                                                     .withRegion(region)
+                                                                     .withExistingResourceGroup(resourceGroup)
+                                                                     .withLeafDomainLabel(azureIPName)
+                                                                     .create());
+
+        // Create a dedicated LB with the required rules
+        LoadBalancer lb = createLoadBalancer(azureService, region, resourceGroup, publicIPAddress);
+
+        // Create the Scale Set (multi-stages)
+        VirtualMachineScaleSet virtualMachineScaleSet = createVMSS(azureService,
+                                                                   region,
+                                                                   resourceGroup,
+                                                                   instance,
+                                                                   network,
+                                                                   lb,
+                                                                   imageNameOrId,
+                                                                   vmAdminUsername,
+                                                                   vmAdminSSHPubKey,
+                                                                   vmAdminPassword,
+                                                                   vmssNbOfInstances,
+                                                                   fileUris,
+                                                                   installCommand);
+
+        logger.info("Azure Scale set '" + azureScaleSetName + "'" + " created inside resource group " + resourceGroup);
+
+        // Return the list of VMs of the Scale Set
+        return virtualMachineScaleSet.virtualMachines()
+                                     .list()
+                                     .stream()
+                                     .map(vm -> instance.withTag(vm.name())
+                                                        .withId(vm.id())
+                                                        .withNumber(SINGLE_INSTANCE_NUMBER))
+                                     .collect(Collectors.toSet());
+    }
+
+    private VirtualMachineScaleSet createVMSS(Azure azureService, Region region, ResourceGroup resourceGroup,
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Method has 13 parameters, which is greater than 7 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS00107)